### PR TITLE
[WIP] Add "theme_experiment" addon support

### PR DESCRIPTION
**Pull request by trickypr**
**What is this**
This is an implementation with the goal of closing #143. You can find more information by reading it.
**tl;dr:** There is a way of distributing themes as extensions to Firefox Nightly and Firefox forks.

**Specific details**
This adds the ability for a user to select the browser they are using and be provided with install instructions for using "theme_experiment" addons within their browser. There is currently no way to have themes that only support "theme_experiment" but themes without "theme_experiment" are not impacted in any way. 

**Checklist**
 - [x] Themes can be added to `themes.json`
 - [x] The tests are updated to ensure they work as intended (depends on [xpi-creator](https://github.com/trickypr/xpi-creator))
 - [x] "theme_experiment" addons have correct install instructions
 - [ ] Method for filtering for only "theme_experiment"
 - [ ] Instructions for theme developers to package
 - [ ] Discussion around the UI / UX of this implementation

Live demo: [https://trickypr.github.io/FirefoxCSS-Store.github.io/](https://trickypr.github.io/FirefoxCSS-Store.github.io/)

### CODE REVIEW

**Comment by BeyondMagic** [comment]
I plan to work on this, but one thing I noticed already is that this PR changes the way we used to build files and it doesn't work properly at all. After #219, I'll take the proper code of this feature and transpile to Astro.

Anyway, thanks @trickypr for working so far in this feature.

**Comment by andreasgrafen** [comment]
I think this actually could be pretty easy – we can tag all themes accordingly and just generate the static files according to wheter the tag is present or not. This would again remove any computational stuff client side.

**Comment by trickypr** [comment]
I probably won't continue working in this. If you want to steal any of my code, feel free.
